keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepositoryTests.kt

```diff
@@ -105,7 +106,13 @@ abstract class ArtifactRepositoryTests<T : ArtifactRepository> : JUnit5Minutests
       .versionsByEnvironment(manifest)
       .first { it.name == environment.name }
       .artifacts
-      .first { it.name == artifact.name && it.type == artifact.type }
+      .first {
+        if (artifact is DebianArtifact) {
+          it.name == artifact.name && it.type == artifact.type && artifact.statuses == it.statuses.toList()
```

Comment:
Should the statuses  comparison be ignoring order? Or perhaps the fields should be sets instead of collections or lists?

Reply:
Great point, will fix.